src/main/java/com/google/starfish/models/Note.java

```diff
@@ -0,0 +1,55 @@
+package com.google.starfish.models;
+
+import java.util.Date;
+
+/** Represents a note */ 
+public class Note {
+
+  private long id;
+  private long authorId;
+  private String school;
+  private String course;
+  private String title;
+  private String sourceUrl;
+  private String pdfSource;
+  private Date dateCreated;
+  private long numDownloads;
+  private long numFavorites;
+  private String[] miscLabels;
+
+  /** Default constructor that throws null exception if NOT NULL fields are null */
+  public Note(long id, 
+      long authorId, 
```

Comment:
I'm not sure what the best style is either, but I was looking around and it seems that some people start each parameter on a new line (so `long id` would be on it's own line) or you could further indent the other parameters so they start where `long id` starts. Just some ideas 😄 , I don't know if you find either of them more readable.

Reply:
These are good suggestions! I did consider putting the first parameter on a new line as well but that seemed to conflict with what I normally do in this case for html tags so I decided against it. Further indentation to align with the first parameter sounds like a good idea but I'm not 100% sure if it would actually really improve readability in this case. Would also love to hear others' thoughts on this!

Reply:
Much less familliar with Java but in Javascript we can do the following:
It might be better to pass in a singular variable that is 1 object containing the initializing properties of a note.

This means that when a caller creates a new note, they won't need to insert undefined or null for unused properties and can just define the ones necessary.

Another option could be to use the builder pattern to construct new notes by adding on properties one at a time.

See here for more details on builder pattern:
https://stackoverflow.com/questions/2432443/best-practice-for-passing-many-arguments-to-method

Reply:
Very interesting, I think the only way to pass a single object containing the initializing properties of the note in java would be to create a separate data abstraction class for the formal parameter list, which wouldn't really make that big of a difference if ALL the parameters were just clumped into one class. An extension of this is to make multiple data abstraction classes that hunk together related parameters but I feel like creating a bunch of these new classes might just clutter up the client code.
I really like the builder pattern and it seems like you can hunk together related parameters even while using the builder pattern just by passing related parameters in the same builder method. I'll push changes that implement the builder pattern on the note and the user since they have so many instance variables.